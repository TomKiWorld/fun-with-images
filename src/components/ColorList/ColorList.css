.colors-list {
  width: 80%;
  max-width: 22rem;
  margin: 0 auto;
  background-color: white;
  padding-top: 1rem;
  font-size: 0.8rem;
}

.colors-list header {
  display: flex;
  align-items: center;
}

.colors-list h2 {
  font-size: 1rem;
  padding: 0 1rem;
  text-align: left;
  margin: 0;
  padding-left: 1.5rem;
}

.colors-list ul {
  text-align: left;
  list-style: none;
  padding: 0 0 0 0.5rem;
  margin-bottom: 0;
}

.colors-list ul li {
  line-height: 2rem;
}

.colors-list ul li span {
  display: inline-block;
  width: 1rem;
  height: 1rem;
  border-radius: 50%;
  margin: -0.2rem 0.5rem;
}

@media screen and (min-width: 31rem) {
  .colors-list {
    font-size: 1rem;
    max-width: unset;
  }

  .colors-list h2 {
    font-size: 1.2rem;
  }
}
