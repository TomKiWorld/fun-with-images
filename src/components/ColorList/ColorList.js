import React from 'react';
import './ColorList.css';

/**
 * Pop up to show colors detected in images
 * 
 * Required props:
 * - colors => Array of colors from the api
 * - onShowClick => On click function to change the value of showColor
 */
const ColorList = ({ colors }) => {
  let colorText = <p>No colors detected.</p>
  if(colors.length) {
    const list = colors.map(color => {
      const { hex, name } = color;
      return <li key={hex} ><span style={{backgroundColor:hex}}></span>HEX: {hex}, Name: {name}</li>;
    });
    colorText = <ul>{list}</ul>;
  }  

  return (
    <article className='colors-list'>
      <h2>Detected colors:</h2>
      {colorText}
    </article>
  );
}

export default ColorList;