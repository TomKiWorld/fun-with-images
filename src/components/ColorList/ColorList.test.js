import { shallow } from 'enzyme';
import React from 'react';
import ColorList from './ColorList';

describe('ColorList Component', () => {
  const mockProps = {
    showColor: true,
    colors: [
      {hex: "#696969", name: "DimGray"},
      {hex: "#bc8f8f", name: "RosyBrown"}
    ]
  };
  const mockShow = () => {
    return mockProps.showColor = !mockProps.showColor;
  };
  const wrapper =shallow(
    <ColorList 
      colors={mockProps.colors} 
      showColor={mockProps.showColor} 
      onShowClick={mockShow}/>
    );
  const colorBtn = wrapper.find('.close-colors').at(0);

  it('Expect to render ColorList Component', () => {     
    expect(wrapper).toMatchSnapshot();
  });

  it('Expect to change showColor', () => {    
    colorBtn.simulate('click');
    expect(mockProps.showColor).toBe(false);
    colorBtn.simulate('click');
    expect(mockProps.showColor).toBe(true);
  });

  it('Expect not to show color ul', () => {
    expect(shallow(<ColorList 
      colors={[]} 
      showColor={mockProps.showColor} 
      onShowClick={mockShow}/>).exists('ul')).toBe(false);
  });

  it('Expect not to show color list container', () => {
    const mockFalseState = {
      showColor: false,
      colors: [
        {hex: "#696969", name: "DimGray"},
        {hex: "#bc8f8f", name: "RosyBrown"}
      ]
    };
    const mockFalseShow = () => {
      return mockFalseState.showColor = !mockFalseState.showColor;
    };
    const wrapper = shallow(
      <ColorList 
        colors={mockFalseState.colors} 
        showColor={mockFalseState.showColor} 
        onShowClick={mockFalseShow}/>
    );
    const colorBtn = wrapper.find('.close-colors').at(0);
    colorBtn.simulate('click');
    expect(mockFalseState.showColor).toBe(true);
    colorBtn.simulate('click');
    expect(mockFalseState.showColor).toBe(false);
  });
});
