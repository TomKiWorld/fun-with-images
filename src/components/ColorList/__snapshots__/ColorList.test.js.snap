// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList Component Expect to render ColorList Component 1`] = `ShallowWrapper {}`;
